threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,55 @@
+| File system     | The infrastructure where Notation is installed, and configuration files are stored and retrieved           |
+| Registry        | A remote registry of Docker and Open Container Initiative (OCI) images, with support for all OCI artifacts |
+| KMS             | A remote Key Management Service for securely storing and accessing secrets                                 |
+
+### Notation sign artifacts using remote key
+
+![image](../media/notation-sign.png)
+
+#### Threat and Mitigation
+
+|   Title                      | STRIDE threat type     | Threat Status | Priority | Entity          | Description   | Mitigation   |
+| ---------------------------- | ---------------------- | ------------- | -------- | --------------- | ------------- | ------------ |
+| Compromised signing key      | Information disclosure | Mitigated     | High     | Signer          | Signing keys are compromised, and this may lead to arbitrary artifacts being signed by attackers | Rotate the signing key, and revoke the associated certificates |
+| Inaccessible Registry        | Denial of Service      | Mitigated     | High     | Registry        | Registry is not able to service incoming requests or perform up to spec, thus users are unable to publish any artifacts | Users sign offline artifacts and publish via other medias |
+| Serving tampered artifacts   | Tampering              | Mitigated     | High     | Registry        | Data stored in registries are tampered, this may lead to malicious artifacts being signed | Users sign offline artifacts, and store signatures locally or in the registry |
+| Disclosure of signature info | Information disclosure | Mitigated     | High     | Registry        | Attackers eavesdrop on registries, this may lead to the disclosure of sensitive signature info | Users should interact with trusted registries via TLS or Users can sign offline artifacts and store signature locally |
+| Tampered files               | Tampering              | Mitigated     | High     | File System     | Files read by Notation are tampered, this may lead to failure of notation sign operation | Use trusted infrastructure with right file permissions |
+| Disclosure of files          | Information disclosure | Mitigated     | High     | File System     | Files are accessed by attackers, this may lead to disclosure of notation config files and credential files | Use trusted infrastructure with right file permissions. Users can use credential store to securely store the credentials. Users can use notation plugin to access signing keys via remote KMS or a HSM device |
+| Malicious plugin             | Tampering              | Mitigated     | High     | Notation Plugin | A malicious plugin is installed, this may lead to arbitrary code being executed by attackers | Verify plugin package before installation or plugin is installed by authorized users |
+
+### Notation verify artifacts against signatures
+
+![image](../media/notation-verify.png)
```

Comment:
For the verification threat model image, since we will also support offline verification, we should also add `Read signatures (from File system)` -> `Notation`. During verification, we still need the target artifact descriptor, thus, we should also add `Read descriptor (from File system)` -> `Notation` and `Fetch descriptor of artifact (from remote registry)` -> `Notation`.